src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversByCarId(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String sql = "SELECT c.id AS car_id, c.model AS car_model, m.id AS manufacturer_id, "
+                + "m.name AS manufacturer_name, m.country AS manufacturer_country, "
+                + "c.is_deleted AS car_is_deleted FROM cars c "
+                + "JOIN manufacturers m ON m.id = manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> allCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allCars.add(parseCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars!", e);
+        }
+        allCars.forEach(c -> c.setDrivers(getDriversByCarId(c.getId())));
+        return allCars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String sql = "UPDATE cars SET manufacturer_id = ?, model = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car + " + car, e);
+        }
+        deleteDriversFromCar(car);
+        updateDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String sql = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String sql = "SELECT car_id FROM cars_drivers JOIN drivers d ON id = driver_id "
+                + "WHERE d.id = ? AND is_deleted = FALSE;";
```

Comment:
isn't it easier to find all users in this query, and then pre-initialize the drivers through a loop?